packages/modals/src/elements/DrawerModal/DrawerModal.tsx

```diff
@@ -0,0 +1,252 @@
+            modalRef.current.style.transform = `translateX(${width}px)`;
+          }
+        }
+      }
+    });
+
+    useEffect(() => {
+      if (modalRef.current) {
+        const { width } = modalRef.current.getBoundingClientRect();
+
+        if (isOpen) {
+          setShowDrawer(true);
+          modalRef.current.style.right = 'initial';
+
+          if (rtl) {
+            modalRef.current.style.left = `-${width}px`;
+            modalRef.current.style.transform = `translateX(${width}px)`;
+          } else {
+            modalRef.current.style.right = `-${width}px`;
+            modalRef.current.style.transform = `translateX(-${width}px)`;
+          }
+        } else {
+          modalRef.current.addEventListener('transitionend', unmountDrawer);
+
+          if (rtl) {
+            modalRef.current.style.transform = 'translateX(0px)';
+          } else {
+            modalRef.current.style.right = `-${width}px`;
+            modalRef.current.style.transform = `translateX(${width}px)`;
+          }
+
+          return () => {
+            modalRef.current &&
+              modalRef.current.removeEventListener('transitionend', unmountDrawer);
+          };
+        }
+      }
+    }, [isOpen, modalRef.current]);
+
+    useEffect(() => {
+      if (!environment) {
+        return undefined;
+      }
+
+      const bodyElement = environment.querySelector('body');
+
+      if (bodyElement && showDrawer) {
+        const previousBodyOverflow = bodyElement.style.overflow;
+
+        bodyElement.style.overflow = 'hidden';
+
+        return () => {
+          bodyElement.style.overflow = previousBodyOverflow;
+        };
+      }
+
+      return undefined;
+    }, [environment, showDrawer]);
+
+    const value = {
+      getTitleProps,
+      getContentProps,
+      getCloseProps
+    };
+
+    return showDrawer || isOpen
+      ? ReactDOM.createPortal(
+          <ModalsContext.Provider value={value}>
+            <StyledDrawerModalBackdrop
+              {...(getBackdropProps({ isAnimated, ...backdropProps }) as any)}
+            >
+              <StyledDrawerModal
+                {...getModalProps({ ref: mergeRefs([ref, modalRef]), ...props } as any)}
+              />
+            </StyledDrawerModalBackdrop>
+          </ModalsContext.Provider>,
+          appendToNode!
+        )
+      : null;
+  }
+) as IStaticDrawerModalExport<HTMLDivElement, IDrawerModalProps>;
+
+DrawerModal.Header = Header;
+DrawerModal.Body = Body;
+DrawerModal.Close = Close;
+DrawerModal.Footer = StyledDrawerModalFooter;
+DrawerModal.FooterItem = StyledDrawerModalFooterItem;
+
+DrawerModal.displayName = 'DrawerModal';
+
+DrawerModal.propTypes = {
+  backdropProps: PropTypes.object,
+  isAnimated: PropTypes.bool,
+  focusOnMount: PropTypes.bool,
+  restoreFocus: PropTypes.bool,
+  onClose: PropTypes.func,
+  appendToNode: PropTypes.any,
+  id: PropTypes.string
+};
+
+DrawerModal.defaultProps = {
+  isAnimated: true,
+  appendToNode: document.body
```

Comment:
I need to update this to reflect https://github.com/zendeskgarden/react-components/pull/876